endpoints/get.go

```diff
@@ -77,31 +74,38 @@ func parseUUID(r *http.Request, allowCustomKeys bool) (string, error) {
 	if uuid == "" {
 		return "", utils.NewPBCError(utils.MISSING_KEY)
 	}
+	// UUIDs are 36 characters long... so this quick check lets us filter out most invalid
+	// ones before even checking the backend.
+	// TODO: allow for custom keys with a length of 36 chars
 	if len(uuid) != 36 && (!allowCustomKeys) {
-		// UUIDs are 36 characters long... so this quick check lets us filter out most invalid
-		// ones before even checking the backend.
 		return uuid, utils.NewPBCError(utils.KEY_LENGTH)
 	}
 	return uuid, nil
 }
 
 // writeGetResponse writes the "Content-Type" header and sends back the stored data as a response if
 // the sotred data is prefixed by either the "xml" or "json"
-func writeGetResponse(w http.ResponseWriter, id string, storedData string) error {
-	if strings.HasPrefix(storedData, backends.XML_PREFIX) {
-		w.Header().Set("Content-Type", "application/xml")
-		w.Write([]byte(storedData)[len(backends.XML_PREFIX):])
-	} else if strings.HasPrefix(storedData, backends.JSON_PREFIX) {
-		w.Header().Set("Content-Type", "application/json")
-		w.Write([]byte(storedData)[len(backends.JSON_PREFIX):])
-	} else {
+func writeGetResponse(w http.ResponseWriter, storedData string) error {
+	switch storedData[0] {
+	case 'x':
+		if strings.HasPrefix(storedData, backends.XML_PREFIX) {
```

Comment:
This `switch` statement looks like an optimization tactic to minimize the number of full string compares. However, as written, I believe if an unknown stored data type is provided that begins with x or j we won't return an error when we should be because the `default` case which returns the error will not be executed.

Reply:
:sweat_smile: Totally agree. Corrected